File: Sources/QuotaGrove/QuotaTheme.swift
import AppKit

enum QuotaTheme: String, CaseIterable, Codable {
    case forest
    case autumn
    case apocalypse
    case wasteland

    static func select(for remainingPercent: Double) -> QuotaTheme {
        switch remainingPercent {
        case 50...100:
            return .forest
        case 20..<50:
            return .autumn
        case 3..<20:
            return .apocalypse
        default:
            return .wasteland
        }
    }

    var accent: NSColor {
        switch self {
        case .forest:
            return NSColor(calibratedRed: 0.47, green: 0.88, blue: 0.67, alpha: 1)
        case .autumn:
            return NSColor(calibratedRed: 0.96, green: 0.72, blue: 0.29, alpha: 1)
        case .apocalypse:
            return NSColor(calibratedRed: 0.94, green: 0.29, blue: 0.28, alpha: 1)
        case .wasteland:
            return NSColor(calibratedRed: 0.78, green: 0.29, blue: 0.28, alpha: 1)
        }
    }

    var borderAccent: NSColor {
        switch self {
        case .wasteland:
            return NSColor(calibratedWhite: 0.94, alpha: 1)
        default:
            return accent
        }
    }

    var accessibilityName: String {
        switch self {
        case .forest: return "额度充足"
        case .autumn: return "额度偏低"
        case .apocalypse: return "额度紧张"
        case .wasteland: return "额度即将耗尽"
        }
    }
}

File: windows/QuotaGrove.Core/QuotaTheme.cs
namespace QuotaGrove.Core;

public enum QuotaTheme
{
    Forest,
    Autumn,
    Apocalypse,
    Wasteland
}

public sealed record QuotaThemeStyle(string AccentHex, string BorderHex, string BackgroundAsset);

public static class QuotaThemes
{
    public static QuotaTheme Select(double remainingPercent) => remainingPercent switch
    {
        >= 70 and <= 100 => QuotaTheme.Forest,
        >= 40 and < 70 => QuotaTheme.Autumn,
        >= 10 and < 40 => QuotaTheme.Apocalypse,
        _ => QuotaTheme.Wasteland
    };

    public static QuotaThemeStyle Style(QuotaTheme theme) => theme switch
    {
        QuotaTheme.Forest => new("#78E0AB", "#6FCF9B", "forest.png"),
        QuotaTheme.Autumn => new("#F5B84A", "#E4A839", "autumn.png"),
        QuotaTheme.Apocalypse => new("#EF4A47", "#E14747", "apocalypse.png"),
        QuotaTheme.Wasteland => new("#C74A47", "#F0F0EA", "wasteland.png"),
        _ => throw new ArgumentOutOfRangeException(nameof(theme))
    };
}
